globals.nix

```diff
@@ -74,7 +74,7 @@ let topologySpec     = builtins.fromJSON (builtins.readFile topologyFile);
     relays          = filter     (x: x.value.typeIsRelay)             indexed;
     nodeMap         = listToAttrs (cores ++ relays);
     # WARNING: this depends on the sort order, as explained above.
-    firstRelay      = findFirst (x: x.value.typeIsRelay) (throw "No relays in topology, it's sad we can't live, but then.. who does?") indexed;
+    firstRelay      = findFirst (x: x.value.typeIsRelay) ({ name = "fake-non-relay"; value = { type = "relay"; i = builtins.length indexed; }; }) indexed;
```

Comment:
why is suddently there a need for fake relay where previously it was an evaluation error?

Reply:
we can evaluate without nodes now

Reply:
But shouldn't we skip the whole topology evaluation if there are no nodes?

Reply:
Ideally, yes.